source/adapters/level_zero/v2/command_list_manager.hpp

```diff
@@ -50,3 +50,8 @@ struct ur_command_list_manager : public _ur_object {
   ur_queue_t_ *queue;
   std::vector<ze_event_handle_t> waitList;
```

Comment:
Is there a way we could make the `waitList` usable only inside of the `getWaitListView` function? Maybe we should store a class here with that member private whose only method is `getWaitListView` ?

Reply:
Yes, but my concern was that the  variable `waitList` was aliased within the scope of a function in a class that already contained `waitList`. Making it inaccessible inside of anything but that function would make it impossible to make a mistake by using the wrong `waitList`.